components/expenses/ExpenseForm.js

```diff
@@ -122,6 +124,26 @@ const ExpenseFormBody = ({ formik, payoutProfiles, collective, autoFocusTitle, o
     }
   }, [payoutProfiles]);
 
+  // Load values from localstorage
+  React.useEffect(() => {
+    if (loggedInAccountId && isEqual(values, initialValues)) {
+      formPersister.setFormId(`expense-${collective.name}-${loggedInAccountId}`);
+    }
+
+    const formValues = formPersister.loadValues();
+    if (formValues && isEqual(values, initialValues)) {
+      setValues(formValues);
+    }
+  }, [loggedInAccountId]);
+
+  // Save values in localstorage
+  React.useEffect(() => {
+    const formValues = formPersister.loadValues();
+    if (!formValues || !isEqual(values, initialValues)) {
```

Comment:
`isEqual` is an expensive operation to do on each change. Is it really necessary?